ts/ui/safe/SafeMethods.ts

```diff
@@ -41,7 +41,7 @@ export const SafeMethods: {[name: string]: FilterFunction<any, any, any>} = {
    * @template D  The Document class
    */
   filterURL<N, T, D>(safe: Safe<N, T, D>, url: string): string | null {
-    const protocol = (url.match(/^\s*([a-z]+):/i) || [null, ''])[1].toLowerCase();
+    const protocol = (url.match(/^\s*([a-z\n\r]+):/i) || [null, ''])[1].toLowerCase();
```

Comment:
Although you know match the newlines, in order to have line 47 below match when they are present, you need to remove them afterward.  For example:

```suggestion
    const protocol = (url.match(/^\s*([a-z\n\r]+):/i) || [null, ''])[1].replace(/[\n\r]/g, '').toLowerCase();
```

That way, looking up the name of the protocol will work even if there were newlines.